### DISCUSSION TITLE
Slow pasting in textarea

### DISCUSSION THREAD

**Discussion by cluster** [charmbracelet/bubbletea#794] [Q&A] [answered]
👋 I am experiencing pretty slow pasting of long texts in a textarea.
I am using the [given example](https://github.com/charmbracelet/bubbletea/blob/master/examples/textarea/main.go)
but I've only updated it to support more characters 
`ti.CharLimit = 0`

![issue_textarea](https://github.com/charmbracelet/bubbletea/assets/167297/55041f63-ce98-497c-903b-ccc1425ed9d7)

When I debug it, it seems that characters are pasted one by one.
Do you know how I can make it faster? 🙃
Thanks for your help

**Comment by meowgorithm** [comment, marked as answer]
Hi! I believe this will be fixed by https://github.com/charmbracelet/bubbletea/pull/397 which we’re actively reviewing — but correct me if I’m wrong @knz / @muesli.

Just out of curiosity, which terminal emulator are you using (as some are known to be faster than others)?

**Reply by cluster** [reply]
Thanks for your reply! 
Once the text is pasted and I start navigating through it with arrow keys, it can get pretty slow too. Do you know if this PR would also fix this?
(I am using the default Terminal application on macos)

**Reply by knz** [reply]
Just curious, how many lines are you pasting before it gets slow?

**Reply by cluster** [reply]
For the pasting, it looks like it's getting slow pretty quickly. I would say that the first 5 lines render quickly, but then it seems that it takes more and more time to render the next character.

Regarding the arrow keys, with about 60 lines, I start experiencing a small lag.

**Reply by knz** [reply]
Yes the pasting situation will be helped like Christian mentioned earlier.

For the arrow keys, there are inefficiencies in textarea which will not go away with that patch. Let's take note of your report at 60 lines.

**Reply by meowgorithm** [reply]
With regard to performance, it’s a known issue. We have a rope implementation planned which should speed it up considerably; see https://github.com/charmbracelet/bubbles/issues/301

**Reply by cluster** [reply]
Great! Thank you both for your answers!
Can't wait to have those PR merged 🙂
